### DISCUSSION TITLE
Directus automatic disconnection since 9.4

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#10726] [Archive]
Before 9.4, I could stay connected indefinitely in Directus. Event overnight, when restarting the computer at morning, I was prompted to clic Continue and went right in were I was.
Since 9.4 I notice that Directus disconnects automatically after a while (15 m. about).
Is there a parameter we can act on?

**Comment by benhaynes** [comment]
Not sure how you installed/upgraded, but are you sure you ran the database migrations?

**Reply by Pachat** [reply]
That's on a new install 
- npm init directus-project myproject
- Choose SQLite
- Sign in and open the Admin user's details
- Just waited 15mn, and without any action from my part, suddenly Directus went back to the login screen.

Also, if I log again, I'm back correctly on the Admin user details.
Now, without waiting like above, Directus shoes to the login screen. instead of the Admin user's details

--

Finally, I opened a project still on 9.3.0 on the same computer, with the same Firefox browser.
- Does not close by itself after 15mn
- Refreshes the page correctly without going back to the login screen

**Reply by benhaynes** [reply]
Interesting. What happens in 9.4 if you hard-refresh the browser _before_ the 15m is up?

**Reply by azrikahar** [reply]
If you have experienced this with SQLite exclusively, then it should be a known bug discovered by @licitdev just earlier today and he pushed a PR #10736 to fix this 👍

**Reply by Pachat** [reply]
@benhaynes:   in 9.4 if i hard-refresh the browser before the 15m is up, i'm disconnected and the login page shows.

**Reply by Pachat** [reply]
9.4.1 resolved the above refresh bug that was present with SQLite. Thanks.

But we still have with SQLite:
- After 15mn, and without any action from my part, suddenly Directus disconnects and goes back to the login screen.
- "Date created", for instance, instantly shows "1 hour ago" (or a past time one hour earlier) °
(This bug I did not mentioned before was already there at least with 9.3.0) 
° My TZ is UTC+1

On the opposite, with MariaDB, none of these bugs are present: everything is OK

**Reply by licitdev** [reply]
Hi @Pachat, thank you for the update.

> * After 15mn, and without any action from my part, suddenly Directus disconnects and goes back to the login screen.

However, I am unable to replicate this, the fix in `9.4.1` resolves the access token refresh issue that you have faced. 

> * "Date created", for instance, instantly shows "1 hour ago" (or a past time one hour earlier)

This is a known issue on date time differences which we're currently looking into: https://github.com/directus/directus/issues/7383

**Reply by Pachat** [reply]
Well, it took me sometime to find out the right scenario, which I must agree is rather weird.

I had on the same Firefox instance, 2 tabs, each running a Directus session each on a distinct installation with a distinct port and a distinct SQLite file : one is running on port 8055 and the other on port 8026
Here is the track from 10:00 to 10:45 a.m.

- at 10:00	
		on port 8026 I log as admin
		on port 8055 I log as admin
		**_on port 8055 I SIGN_OUT_**
 
- at 10:15	
		**_on port 8026 SESSION_EXPIRED (automatically)_**
		on port 8026 I log  as admin
 
- at 10:30	
		on port 8026 still logged
		on port 8055 I log  as admin
		**_on port 8055 I SIGN_OUT_**
 
- at 10:45	
		**_on port 8026 SESSION_EXPIRED (automatically)_**

So, as you can see, when I log on 8055 then immediately log out at 10:00, that 8026 logs out automatically at 10:20
Then, when I log on 8026 at 10:20, but do nothing on 8055, 8026 stay logged at 10:40
Finally, as a verification,  I log on 8055 then immediately log out at 10:40, and 8026 logs out automatically at 11:00

This is may be due to a shared cookie between the two sessions?

May be related to  #10654

**Reply by licitdev** [reply]
> This is may be due to a shared cookie between the two sessions?

@Pachat That should be it, the cookies are shared through the domain name. You may wish to set a different `REFRESH_TOKEN_COOKIE_NAME` for each Directus instance. https://docs.directus.io/configuration/config-options/#security

**Reply by rijkvanzanten** [reply]
cc @erondpowell

**Comment by Pachat** [comment]
> > This is may be due to a shared cookie between the two sessions?
> 
> @Pachat That should be it, the cookies are shared through the domain name. You may wish to set a different `REFRESH_TOKEN_COOKIE_NAME` for each Directus instance. https://docs.directus.io/configuration/config-options/#security

@licitdev Yes. That's it! Thanks.
